transport/transport.go

```diff
@@ -341,8 +341,10 @@ func NewServerTransport(protocol string, conn net.Conn, maxStreams uint32, authI
 type ConnectOptions struct {
 	// UserAgent is the application user agent.
 	UserAgent string
-	// Dialer specifies how to dial a network address.
-	Dialer func(string, time.Duration) (net.Conn, error)
+	// Cancel is closed to indicate that dialing should be cancelled.
+	Cancel chan struct{}
+	// dialer specifies how to dial a network address.
+	dialer func(string, time.Duration, <-chan struct{}) (net.Conn, error)
```

Comment:
> It's left unexported to encourage users to use the `Dial` method rather
> than accessing this directly. That method encapsulates usage of the default.

The only "user" is newHTTP2Client(...). This is not a public facing API -- we only need to make the only caller correct.